pipelines/batch/src/main/java/com/google/fhir/analytics/FhirEtl.java

```diff
@@ -377,20 +391,28 @@ private static List<Pipeline> buildParquetReadPipeline(
     return pipelineList;
   }
 
-  private static List<Pipeline> buildJsonReadPipeline(FhirEtlOptions options) throws IOException {
+  private static List<Pipeline> buildJsonReadPipeline(FhirEtlOptions options) {
+    PCollection<FileIO.ReadableFile> files =
+        buildPipelineForFilePattern(options, options.getSourceJsonFilePattern());
+    files.apply("Read JSON files", ParDo.of(new ReadJsonFilesFn(options)));
+    return Arrays.asList(files.getPipeline());
```

Comment:
In this and the next function, is there a particular reason to go from a `PCollection` to a `List`? In general, I try to avoid such conversions if possible, i.e., stay in the non-parallel collections world at the start of the pipelines and once starting to build parallel collections (e.g., `PCollection`) stay within Beam's paradigm.

Reply:
The method has been refactored to avoid such confusions.